src/managers/socketManager.ts

```diff
@@ -1,33 +1,32 @@
-import socketIo from "socket.io";
-import { createServer, Server } from "http";
 import { app, dialog } from "electron";
-import { update as updateSettings } from "./settingsManager";
+import { createServer, Server } from "http";
+import socketIo from "socket.io";
+
+import { trayManager } from "../";
+import { clearActivity, getDiscordUser, rpcClients, setActivity } from "./discordManager";
 import { openFileDialog } from "./presenceDevManager";
-import {
-	rpcClients,
-	setActivity,
-	clearActivity,
-	getDiscordUser
-} from "./discordManager";
-import { trayManager } from "..";
+import { update as updateSettings } from "./settingsManager";
 
 export let io: socketIo.Server;
 export let socket: socketIo.Socket;
 export let server: Server;
 export let connected: boolean = false;
 
 export function init() {
-	return new Promise(resolve => {
+	return new Promise<void>(resolve => {
 		//* Create server
 		//* create SocketIo server, don't server client
 		//* Try to listen to port 3020
 		//* If that fails/some other error happens run socketError
 		//* If someone connects to socket socketConnection
 		server = createServer();
-		io = socketIo(server, { serveClient: false });
-		io.origins((o, c) => {
-			if (o !== "*") c("Not a chrome extension, socket denied.", false);
-			c(null, true);
```

Comment:
Wasnt this needed because of your old CVE?

Reply:
DId you read this code or not? Because I literally included stuff from my pull request @ https://github.com/PreMiD/PreMiD/pull/791 
![image](https://user-images.githubusercontent.com/9348108/146677068-296c8ab2-cdc8-4b60-b979-776208868d78.png)

Reply:
`!== "*"` !== `=== undefined`
Hence why I asked

Reply:
If you don’t understand still then lemme help you.
socket.io v2 was treating request that didn’t have origin header as “*” so that’s why it was checking if it isn’t “*”. v3 perhaps is just strictly giving you option to check if it is undefined lol. This is the same code as in socket.io v2 but reworked for v3.

Reply:
Next time please read docs before asking.

Reply:
Or in this comment I explained that it has been removed when there was migration to v3 https://github.com/PreMiD/PreMiD/issues/790#issuecomment-997281008